### ISSUE TITLE
Replace ButtonGroup usage with ToggleGroupControl

### ISSUE THREAD

**Issue by mirka** [WordPress/gutenberg#65339]
Part of #65338

All of our [remaining usage](https://github.com/search?q=repo%3AWordPress%2Fgutenberg+%3Cbuttongroup+-path%3A%2F%5Epackages%5C%2Fcomponents%5C%2Fsrc%5C%2F%2F&type=code) of the ButtonGroup component is visually outdated as a segmented control and often inaccessibly implemented. We should correct this by replacing them with [ToggleGroupControl](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-togglegroupcontrol--docs).

## How to submit PRs

To pick up a task, write your name by the checklist item or post a comment to this issue.

Please include before/after screenshots in the PR.

- [x] `packages/block-editor/src/components/image-size-control/index.js` #65386
- [x] `packages/block-library/src/button/edit.js` - #65346
- [x] `packages/block-editor/src/hooks/layout.js` - #65498
- [x] `packages/block-library/src/search/edit.js` - #65340

**Comment by hbhalodia** [comment]
Hi @mirka, I want to work on the first issue mentioned in the description.

Thank You,

**Comment by hbhalodia** [comment]
Hi @mirka - Have raised the PR for the task 1 - https://github.com/WordPress/gutenberg/pull/65386

Thank You,

**Comment by spadeshoe** [comment]
@mirka I would like to work on the packages/block-editor/src/hooks/layout.js part of this issue. I am unsure of what component in the Editor LayoutTypeSwitcher is affecting so I have been unable to test that my change has not broken anything. I thought it might be the type size selector. See screenshot.

<img width="331" alt="Screenshot 2024-09-17 at 2 47 30 PM" src="https://github.com/user-attachments/assets/a5e1d9b2-5991-4f10-ab2e-e1d5aa44cbfe">

**Comment by mirka** [comment]
@spadeshoe Good question! I don't think this code path is actually reached in any of our core blocks. So you can force the condition like this:

```diff
diff --git a/packages/block-editor/src/hooks/layout.js b/packages/block-editor/src/hooks/layout.js
index 22d916d7b7..7a349ceb0b 100644
--- a/packages/block-editor/src/hooks/layout.js
+++ b/packages/block-editor/src/hooks/layout.js
@@ -264,7 +264,7 @@ function LayoutPanelPure( {
 						</>
 					) }
 
-					{ ! inherit && allowSwitching && (
+					{ true && (
 						<LayoutTypeSwitcher
 							type={ blockLayoutType }
 							onChange={ onChangeType }

```

Then insert a Grid block, for example. And you'll see this:

<img src="https://github.com/user-attachments/assets/2849a28f-a8a5-436f-be2f-514c2425fa8c" alt="Layout type switcher" width="271">

**Comment by spadeshoe** [comment]
Thanks @mirka The conditional trick worked and I was able to test the update. I created the PR for the change: [https://github.com/WordPress/gutenberg/pull/65498](https://github.com/WordPress/gutenberg/pull/65498)

@jeryj and I had done some digging Tuesday and discovered what you did, that this code is no longer used in core blocks. I have a branch that removes this code all together. I'm not sure what the protocol is for removing "deprecated" code, if it should be removed at all for backwards compatibility.

**Comment by mirka** [comment]
@spadeshoe Thanks for the PR, we'll get it reviewed shortly!

> I had done some digging Tuesday and discovered what you did, that this code is no longer used in core blocks. I have a branch that removes this code all together. I'm not sure what the protocol is for removing "deprecated" code, if it should be removed at all for backwards compatibility.

Although it isn't used in any of the blocks in core, it's still a [supported block API](https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/block-api/block-supports.md#layoutallowswitching) so we'll need to keep it 👍

**Comment by t-hamano** [comment]
I believe all tasks are complete and would like to close this issue.

**Comment by ciampo** [comment]
@mirka , anything left before being able to deprecate `ButtonGroup` ?


Edit: nevermind, just seen #65429

**Comment by mirka** [comment]
Last instance that I somehow missed: #67985
